app/controllers/users_controller.rb

```diff
@@ -15,9 +15,17 @@ def admin_index
     authorize User
     if current_user.can_super_admin?
```

Comment:
I'd propose doing something like this to prevent the code from running both queries for all requests:
```ruby
case request.format
    when 'text/csv' # there might be a better way to test for the format
      if current_user.can_super_admin?
        @users = User.all.order(:surname)
      else
        current_user.org.users.order(:surname)
      end
    else
      if current_user.can_super_admin?
        @users = User.page(1)
      else
        @users = current_user.org.users.page(1)
      end
    end
```

Reply:
@briri Updated code. I have excluded super admin user viewing the *Download Csv* button.

Reply:
@briri & @Bodacious : I based my changes on comments from both. All mistakes are mine.